eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServer.java

```
@@ -55,98 +64,70 @@ public RegionWSServer() {
-                                             @RequestParam(name = "polyphen", defaultValue = "") String polyphenScore,
-                                             @RequestParam(name = "sift", defaultValue = "") String siftScore,
-                                             @RequestParam(name = "ref", defaultValue = "") String reference,
-                                             @RequestParam(name = "alt", defaultValue = "") String alternate,
-                                             @RequestParam(name = "miss_alleles", defaultValue = "") String missingAlleles,
-                                             @RequestParam(name = "miss_gts", defaultValue = "") String missingGenotypes,
-                                             @RequestParam(name = "histogram", defaultValue = "false") boolean histogram,
-                                             @RequestParam(name = "histogram_interval", defaultValue = "-1") int interval,
-                                             @RequestParam(name = "merge", defaultValue = "false") boolean merge,
+                                             @RequestParam(name = "maf", defaultValue = "", required = false) String maf,
+                                             @RequestParam(name = "polyphen", defaultValue = "", required = false) String polyphenScore,
+                                             @RequestParam(name = "sift", defaultValue = "", required = false) String siftScore,
                                              HttpServletResponse response)
             throws IllegalOpenCGACredentialsException, IOException {
         initializeQueryOptions();
 
-        VariantDBAdaptor variantMongoDbAdaptor = DBAdaptorConnector.getVariantDBAdaptor(species);
+        if (species == null || species.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setQueryResponse("Please specify a species");
+        }
```

Then the check null is not needed, and my question still stands as how to proceed with error and if we should throw a exceptions as you do at line 125.